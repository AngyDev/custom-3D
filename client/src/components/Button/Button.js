import React from "react";
import PropTypes from "prop-types";

export default function Button({ typeClass, type, onClick, title, disabled, text, img, active, imgRight }) {
  return (
    <button
      className={`btn ${typeClass} ${disabled ? "bg-gray-500 cursor-not-allowed disabled:bg-gray-500" : ""} ${active ? "bg-yellow-600" : ""}`}
      type={type}
      onClick={onClick}
      title={title}
      disabled={disabled}
    >
      {img && !imgRight && <img className="w-4 h-4" src={img} alt={title} />}
      {text && <span className="pl-2 pr-2">{text}</span>}
      {imgRight && <img className="w-4 h-4" src={img} alt={title} />}
    </button>
  );
}

Button.propTypes = {
  typeClass: PropTypes.string,
  type: PropTypes.string,
  onClick: PropTypes.func,
  title: PropTypes.string,
  disabled: PropTypes.bool,
  active: PropTypes.bool,
  text: PropTypes.string,
  img: PropTypes.string,
  imgRight: PropTypes.bool,
};
