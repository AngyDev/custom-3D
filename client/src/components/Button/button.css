/* .btn {
  @apply bg-gray-400 text-gray-500 border-0 rounded m-0 pt-1 pl-2 pb-1 pr-2 uppercase cursor-pointer outline-none relative;
}

.btn--size {
  @apply text-sm;
}

.btn--img {
  @apply flex;
}

.btn__img {
  @apply w-4 h-4 items-center justify-center;
} */

.btn {
  @apply border-none flex justify-center items-center bg-gray-400 text-black font-bold m-0 cursor-pointer outline-none uppercase rounded p-2 hover:bg-slate-500;
}

.btn-active {
  @apply bg-slate-500;
}

/* .btn--img {
  @apply flex;
} */

.btn--size {
  @apply text-sm;
}

input[type="reset"] {
  @apply bg-gray-200;
}