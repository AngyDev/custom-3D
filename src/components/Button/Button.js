import React from 'react'

export default function Button(props) {
    return (
        <button
            className={`btn ${props.typeClass}`}
            type={props.type}
            disabled={props.loading}
            onClick={props.onClick}
        >
            {props.text}
        </button>
    )
}
