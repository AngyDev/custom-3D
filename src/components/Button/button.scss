.btn {
    color: #555;
    background-color: #ddd;
    border: 0px;
    margin: 0px;
    padding: 5px 8px;
    text-transform: uppercase;
    cursor: pointer;
    outline: none;
    position: relative;
}
