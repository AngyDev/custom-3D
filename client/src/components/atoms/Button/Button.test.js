import React from "react";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import Button from "./Button";

describe("Button component", () => {
  test("Renders the button component without crashing", () => {
    render(<Button />);

    expect(screen.getByRole("button")).toBeInTheDocument();
  });

  test("Renders the button with props", () => {
    render(<Button text="Click" type="button" disabled={true} />);

    expect(screen.getByRole("button")).toBeInTheDocument();
    expect(screen.getByRole("button")).toHaveTextContent("Click");
    expect(screen.getByRole("button")).toBeDisabled();
  });

  test("Renders the button with image", () => {
    render(<Button text="Click" type="button" img="img" title="image" />);

    const image = screen.getByAltText("image");

    expect(image).toHaveAttribute("src", "img");
  });

  test("matches snapshot", () => {
    const tree = renderer.create(<Button text="Click" type="button" img="img" title="image" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
