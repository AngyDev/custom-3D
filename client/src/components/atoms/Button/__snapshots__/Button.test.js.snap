// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches snapshot 1`] = `
<button
  className="btn undefined  "
  title="image"
  type="button"
>
  <img
    alt="image"
    className="w-4 h-4"
    src="img"
  />
  <span
    className="pl-2 pr-2"
  >
    Click
  </span>
</button>
`;
